packages/composite-checkout/src/components/billing-fields.js

```diff
@@ -0,0 +1,552 @@
+/**
+ * External dependencies
+ */
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+
+/**
+ * Internal dependencies
+ */
+import { useLocalize } from '../lib/localize';
+import { useCheckoutLineItems } from '../index';
+import GridRow from './grid-row';
+import Field from './field';
+
+export default function BillingFields( {
+	summary,
+	setPaymentData,
+	paymentData,
+	isActive,
+	isComplete,
+} ) {
+	const [ items ] = useCheckoutLineItems();
+	const [ showDomainContactFields, setShowDomainContactFields ] = useState( false );
+
+	if ( summary && isComplete ) {
+		return <BillingFormSummary paymentData={ paymentData } />;
+	}
+	if ( ! isActive ) {
+		return null;
+	}
+
+	function toggleDomainFieldsVisibility() {
+		setShowDomainContactFields( showDomainContactFields ? false : true );
+	}
+
+	return (
+		<BillingFormFields>
+			{ hasDomainsInCart( items ) && (
+				<AddressFields
+					paymentData={ paymentData }
+					setPaymentData={ setPaymentData }
+					location={ 'billing' }
+				/>
+			) }
+
+			<TaxFields
+				paymentData={ paymentData }
+				setPaymentData={ setPaymentData }
+				location={ 'billing' }
+			/>
+
+			<PhoneNumberField
+				paymentData={ paymentData }
+				setPaymentData={ setPaymentData }
+				location={ 'billing' }
+			/>
+
+			{ isElligibleForVat() && (
+				<VatIdField setPaymentData={ setPaymentData } paymentData={ paymentData } />
+			) }
+
+			{ hasDomainsInCart( items ) && (
+				<DomainRegistrationCheckBoxWrapper>
```

Comment:
I wanted to to use [this component](https://github.com/Automattic/wp-calypso/pull/37006/files#diff-c5f84efd7083bab6ff782c0836c84317R117) and pass the `toggleDomainFieldsVisibility` method as the `toggleVisibility` parameter but it threw an error saying it was an object not function. Any ideas why?

Reply:
aaah yes! Ugg, I'm still getting used to this new syntax. I made the same mistake in my payment methods PR but then figured out why it wasn't working. Thanks for pointing that out!